tests/core/data_layer/test_data_store.py

```diff
@@ -1137,11 +1137,24 @@ async def test_change_root_state(data_store: DataStore, tree_id: bytes32) -> Non
     )
     root = await data_store.get_pending_root(tree_id)
     assert root is not None
+    assert root.status == Status.PENDING
+    is_empty = await data_store.table_is_empty(tree_id=tree_id)
+    assert is_empty
+
+    await data_store.change_root_status(root, Status.PENDING_BATCH)
```

Comment:
I think it would be nice to have a separate test (or two more?) for granular results.